src/lxc/lxccontainer.c

```diff
@@ -4695,14 +4695,14 @@ static int do_lxcapi_mount(struct lxc_container *c,
 		return -EINVAL;
 	}
 
-	len = strlen(c->lxc_conf->lxc_shmount.path_host) + sizeof("/lxcmountXXXXXX");
-	template = malloc(len);
+	len = strlen(c->lxc_conf->lxc_shmount.path_host) + sizeof("/lxcmountXXXXXX") - 1;
+	template = malloc(len + 1);
 	if (!template) {
 		SYSERROR("Not enough memory");
 		return -ENOMEM;
 	}
-	ret = snprintf(template, len, "%s/lxcmountXXXXXX", c->lxc_conf->lxc_shmount.path_host);
-	if (ret < 0 || (size_t)ret >= len) {
+	ret = snprintf(template, len + 1, "%s/lxcmountXXXXXX", c->lxc_conf->lxc_shmount.path_host);
```

Comment:
> If `len` does include the count for the `\0` byte you may get truncated results although it didn't have to be that way.
Why? It would be truncated if we ask for a too small limit, i.e. `len` not including `\0`? 

But I changed the `len` so it is now strlen-like :)

Reply:
Nice.